runtime/autoload/man.vim

```diff
@@ -2,15 +2,24 @@
 
 let s:man_find_arg = "-w"
 
-" TODO(nhooyr) Completion may work on SunOS; I'm not sure if `man -l` displays
-" the list of searched directories.
-try
-  if !has('win32') && $OSTYPE !~? 'cygwin\|linux' && system('uname -s') =~? 'SunOS' && system('uname -r') =~# '^5'
-    let s:man_find_arg = '-l'
-  endif
-catch /E145:/
-  " Ignore the error in restricted mode
-endtry
+function s:init() abort
+  " TODO(nhooyr) Completion may work on SunOS; I'm not sure if `man -l` displays
+  " the list of searched directories.
+  try
+    if !has('win32') && $OSTYPE !~? 'cygwin\|linux' && system('uname -s') =~? 'SunOS' && system('uname -r') =~# '^5'
+      let s:man_find_arg = '-l'
+    endif
+    let s:man_has_l_flag = 1
+    call s:get_page(s:get_path('', 'man'))
+  catch /E145:/
+    " Ignore the error in restricted mode
+  catch /command error .*/
+    let s:man_has_l_flag = 0
+  catch
+    let s:man_has_l_flag = 0
```

Comment:
I guess that depends your the perspective. Not-zero was the default behavior of this plugin (so I kept it that way) for Mac and some other BSDs zero is the value that you need otherwise `:Man` won't work properly.